src/internal/analytics/components/analytics-funnel.tsx

```diff
@@ -68,6 +68,8 @@ const InnerAnalyticsFunnel = ({ children, stepConfiguration, ...props }: Analyti
   const [funnelInteractionId, setFunnelInteractionId] = useState<string>('');
   const [submissionAttempt, setSubmissionAttempt] = useState(0);
   const isVisualRefresh = useVisualRefresh();
+  const inheritedFunnelNameSelector = useFunnelNameSelector();
+  const funnelNameSelector = props.funnelNameSelector || inheritedFunnelNameSelector || getFunnelNameSelector();
```

Comment:
Because `getFunnelNameSelector` is a function returning a constant value, I would recommend refactoring it into a plain constant.

Current state makes it prone to having side effects inside React's render function, possibly returning incorrect selectors